### PR TITLE
Update accordion to use HTML5 elements

### PR DESCRIPTION
**Pull request by maya**
This has some breaking changes so possibly would make more sense for 2.0.

[😎 Preview](https://federalist-proxy.app.cloud.gov/preview/uswds/uswds/update-accordion-html5/components/preview/accordion--default.html)

Related: #2049.

### CODE REVIEW

**Comment by jordangov** [comment]
Hmm... I'm not sure what that specific warning is for, but I do see that there isn't a single node for each accordion panel, which I think is going to be necessary. Basically, the `{% for item in accordion.items %}` needs a single element inside of it that represents the entire accordion panel?

**Comment by maya** [comment]
The error is referring to this ["kitchen sink" page](https://federalist-proxy.app.cloud.gov/preview/uswds/uswds/update-accordion-html5/components/preview/kitchen-sink.html) and is saying "you can't have a `<main>` inside a `<main>`" since the main content of the page is wrapped in a `<main>`. Is that error accurate?

**Comment by maya** [comment]
>Basically, the {% for item in accordion.items %} needs a single element inside of it that represents the entire accordion panel?
Hmm... I didn't see each panel wrapped in an element in Heydon's example: https://codepen.io/heydon/pen/QqzRvQ

**Comment by joshbruce** [comment]
Can a page have more than one `main`? Thought this was going to use the `summary` + `details`.

**Comment by joshbruce** [comment]
Oh! My bad @Maya - was looking at this on my phone - thought each accordion was a `main` for some reason. I dig it. 

In case you didn't see this conversation before - not sure if there's anything that might change anything here: https://github.com/uswds/uswds/issues/1136

**Comment by maya** [comment]
>Thought this was going to use the `summary` + `details`.
I would love to use that element. The problem is I don't really know the state of how accessible it is. Would we ultimately have to write custom JS to make it accessible? So rather than getting the accessibility of a native button like here, we'd have to write JS to do that? So you end up either having to write JS to open and close it and get a11y for free, or have it open + close natively and write JS to make it accessible. We'd also need to provide a polyfill as it looks like it's not supported in any IE.

Thanks for linking to that issue, as you can see in here, we're not using lists for it :)

**Comment by maya** [comment]
Also, posting this to look at: https://scottaohara.github.io/accessible_accordions/

**Comment by maya** [comment]
@jordangov I think `<header>` would have to be changed to a heading level  (e.g. h2) to be more accurate. I worry it could be misused with copy + paste, so we can add a comment in the code to alleviate this.

**Comment by jordangov** [comment]
sure, I think we're on the right track... and anything is better than div's ;)

**Comment by joshbruce** [comment]
I feel like we’ve been here before, but I missed where we went a different route. Seriously getting a sense of de ja vu.

The JS used to run on load to set `aria-hidden` to true, which handles the accordions expanded by default thing, because they would collapse when JS ran...think I actually did a PR for something related to it; could be wrong.

Then @maya and I played a PR iteration dance that was fun related to an `aria-hidden` thing.

Glad to the `div` and list go away and trying to get back to possibly a more simplified implementation.

**Comment by maya** [comment]
It sounds familiar, but when I ran the accordions with JS turned off, they're all closed.

**Comment by joshbruce** [comment]
Yeah? 

So weird. Check this file...and related things (including CSS): https://github.com/joshbruce/web-design-standards/blob/staging/src/js/components/accordion.js

For the CSS, check PR history. Their was like three of us involved. A user with”duck” in his handle is coming to mind.

And all the conversation re `aria-hidden` and how it shouldn’t be associated with CSS - specifically `display: none`.

**Comment by joshbruce** [comment]
Feels like JS maybe started expanding a bit more than it maybe needed to (compared to progressive enhancement)?? 

But I wasn’t paying attention to the conversations that got that going...this general trend on the web is partially the inspiration behind: https://m.8fold.pub/on-constraints-internet-bandwidth-eab05c20e218?source=linkShare-f4442260749b-1518840173 - shameless plug...a little sorry about that.

File: src/components/06-accordion/accordion.njk [hunk 1]
```diff
@@ -1,16 +1,17 @@
...
       aria-controls="{{ accordion.id_prefix }}{{ item.id }}">
       {{ item.title }}
     </button>
-    <div id="{{ accordion.id_prefix }}{{ item.id }}" class="usa-accordion-content">
-      {{ item.content }}
-    </div>
-  </li>
+  </h2>
+  <div id="{{ accordion.id_prefix }}{{ item.id }}" class="usa-accordion-content">
+    {{ item.content }}
```
Comment (donjo) [inline src/components/06-accordion/accordion.njk]: In an effort to keep HTML tags out of the config we should probably move the `<p>` here to wrap {{ item.content }}

Comment (maya) [reply src/components/06-accordion/accordion.njk]: I'm testing other text elements (e.g. headings and lists) to ensure resilience and flexibility in the accordion panels, so keeping the text in the config for now.

**Comment by maya** [comment]
Merging into 2.0 branch... _bombs away_

**Pull request merged**